### ISSUE TITLE
Focus issue in ToolbarSearch when typing search text

### ISSUE THREAD

**Issue by amuqeetkhan** [carbon-design-system/carbon#5852]
## Environment

> Operating system: Linux

> Browser: Firefox, Chrome

## Detailed description
We have an issue with `ToolbarSearch` where when we type some text we cannot go out of the text box using tab until we clear the text box.

We also cannot tab to the `x` icon to clear the text, we have to use backspace. Then we can tab out of the search text box to other elements

> What version of the Carbon Design System are you using?
Carbon v10
carbon-components=10.10.1
carbon-components-react=7.10.1

> What did you expect to happen?
Hitting tab when typing something inside `ToolbarSearch` should take us to next element

> What happened instead?
It is stuck until we use backspace to clear text. Then only we can tab out of the box

## Steps to reproduce the issue

1. Tab into `ToolbarSearch`
2. Type some text
3. Then hit tab, the focus is stuck on the box
4. Clear the text using backspace and now hit tab, we can move to next element

**Comment by amuqeetkhan** [comment]
@joshblack @asudoh
